platforms/chibios/interrupt_handlers.c

```diff
@@ -0,0 +1,42 @@
+// Copyright 2023 Nick Brassel (@tzarc)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+///////////////////////////////////////////////////////////////////////////////
+// BEGIN: STM32L4xx Wear-leveling ECC fault handling
+//
+// STM32L4xx have ECC checks for all flash memory access. Whenever there's an
+// ECC failure, the MCU raises the NMI interrupt -- in ChibiOS this is normally
+// used for context switching, but in the case of L4xx we need to ensure the
+// method used for context switching does not collide with the NMI interrupt.
```

Comment:
Apparently ChibiOS uses NMI for context switching only on ARMv6 (Cortex-M0[+]/M1), unless `#define CORTEX_ALTERNATE_SWITCH TRUE` is used — in that case the PendSV vector is used instead on NMI.  On ARMv7 context switching uses the SVC vector by default, or the PendSV vector if `#define CORTEX_SIMPLIFIED_PRIORITY TRUE` is used.  So apparently L4xx does not require any extra configuration, because it's ARMv7.

Reply:
Yeah this comment is left over from when it wasn't limited to L4xx. Will re-word it.